src/target/source/codegen_c_host.cc

```diff
@@ -355,22 +374,24 @@ runtime::Module BuildCHost(IRModule mod, Target target) {
   Map<String, LinkedParam> linked_params;
   PrimFunc aot_executor_fn;
 
+  std::vector<std::pair<tvm::GlobalVar, tvm::BaseFunc>> funcs;
   for (auto kv : mod->functions) {
```

Comment:
now this logic is a bit confusing, because it's already trying to reorder the functions. i think it might make sense to keep all the ordering logic in one spot. maybe suggest to just put it all in here, since we already have AddFunction as the helper function?

Reply:
done.